src/db.c

```diff
@@ -1314,7 +1314,7 @@ void typeCommand(client *c) {
     addReplyStatus(c, getObjectTypeName(o));
 }
 
-/* SHUTDOWN [[NOSAVE | SAVE] [NOW] [FORCE] | ABORT] */
+/* SHUTDOWN [[NOSAVE | SAVE] [NOW] [FORCE] [SAFE] | ABORT] */
```

Comment:
Can we forbid the comination FORCE and SAFE?

```suggestion
/* SHUTDOWN [[NOSAVE | SAVE] [NOW] [FORCE | SAFE] | ABORT] */
```

Also in json file and in the check for illegal combo below.

Reply:
I somehow want to keep it, the safe thing is just like another error that we can think? I am ok with both anyway, forbid also seem reasonable
```c
            if (force) {
                serverLog(LL_WARNING, "Error trying to save the DB. Exit anyway.");
            } else {
                serverLog(LL_WARNING, "Error trying to save the DB, can't exit.");
```

Reply:
I'm just thinking about the case someone uses FORCE and SAFE together by mistake. The server exits even if it's not safe?

Reply:
I mentioned it in the conf
```
# safe:     Shutdown only when safe. Note that safe cannot prevent force, in the case of
#           force, safe will print the relevant logs. The definition of safe may be different
#           in different modes. Here are the definitions:
#           1. In cluster mode, it is unsafe to shutdown a primary with slots, and may cause
#              the cluster to go down.
```

|           | not SAFE | SAFE |
|-----------|----------|------|
| not FORCE | pass        | dont exit if not safe, exit only safe    |
| FORCE     | exit anyway        | exit anyway but print a log if unsafe    |

Reply:
The "exit anyway but print a log if unsafe" seems risky. Maybe we can do "don't exit" in this case.. If they want to exit, they need FORCE without SAFE. Seems safer?

Reply:
I guess FORCE + SAFE can still be useful. It can shut down a replica when save dump to disk failed.

Reply:
> I guess FORCE + SAFE can still be useful. It can shut down a replica when save dump to disk failed.

FORCE can anyway ignore a dump failure and exit right now (without SAFE)

Reply:
Do you want to forbid it? I think FORCE can anyway shutdown the node, but in some cases we may print some useful logs. I am ok with both, i actually forbid it the first time, but when i update isValidShutdownOnSigFlags, i thought again, why not allow them to be combined? haha

Reply:
I'm OK with your idea.